sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/TracerProperties.java

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.tracing;
+
+/**
+ * A generic interface for setting tracing properties on outgoing HTTP requests.
+ */
+public interface TracerProperties {
```

Comment:
Can this be marked as a `@FunctionalInterface`? If yes, can we instead use Java's built-in Supplier?

Reply:
updated it to be  a supplier.